lib/matplotlib/collections.py

```diff
@@ -476,7 +480,15 @@ def set_linewidth(self, lw):
         ACCEPTS: float or sequence of floats
         """
         if lw is None:
-            lw = mpl.rcParams['patch.linewidth']
+            if (self._edge_default or
+                mpl.rcParams['_internal.classic_mode'] or
+                not self._is_filled):
```

Comment:
No one has ever shown me a convincing alternative that doesn't look worse or more confusing, to be honest.